Integrations/FortiSIEM/FortiSIEM.py

```diff
@@ -0,0 +1,690 @@
+        "selectClause": "phRecvTime,reptDevIpAddr,eventType,eventName,rawEventMsg,destIpAddr",
+        "reportWindow": int(reportWindow),
+        "reportWindowUnit": reportWindowUnit,
+        "timeRangeRelative": True,
+        "eventFilters": [{
+            "groupBy": "",
+            "singleConstraint": query_string
+        }],
+        "custId": 1
+    }
+    return getEventsByQuery(
+        session,
+        query_data,
+        maxResults,
+        extendedData,
+        maxWaitTime,
+        "FortiSIEM Event Results")
+
+
+def parse_cmdb_list(cmdb_device):
+    device_dict = {
+        'DiscoverMethod': cmdb_device.get('discoverMethod', 'N/A'),
+        'Approved': cmdb_device.get('approved', 'false'),
+        'CreationMethod': cmdb_device.get('creationMethod', 'N/A'),
+        'AccessIp': cmdb_device.get('accessIp', 'N/A'),
+        'Name': cmdb_device.get('name', 'N/A'),
+        'WinMachineGuid': cmdb_device.get('winMachineGuid', 'N/A'),
+        'Unmanaged': cmdb_device.get('unmanaged', 'false'),
+        'Version': cmdb_device.get('version', 'N/A'),
+        'UpdateMethod': cmdb_device.get('updateMethod', 'N/A'),
+    }
+    timestamp = cmdb_device.get('discoverTime', None)
+
+    if timestamp and timestamp.isdigit():
+        device_dict['DiscoverTime'] = timestamp_to_datestring(timestamp)
+    elif timestamp:
+        device_dict['DiscoverTime'] = timestamp
+    else:
+        device_dict['DiscoverTime'] = 'N/A'
+
+    device_type = cmdb_device.get('deviceType')
+    if device_type:
+        device_dict['DeviceType'] = "{} {}".format(device_type['model'], device_type['vendor'])
+    else:
+        device_dict['DeviceType'] = 'N/A'
+
+    return device_dict
+
+
+def get_cmdb_devices_command():
+    args = demisto.args()
+    device_ip = args.get('device_ip')
+    limit = int(args.get('limit'))
+
+    raw_response = get_cmdb_devices(device_ip, limit)
+    list_of_devices = list(map(parse_cmdb_list, raw_response))
+
+    return_outputs(
+        tableToMarkdown("Devices", list_of_devices),
+        {'FortiSIEM.CmdbDevices': list_of_devices},
+        list_of_devices
+    )
+
+
+def get_cmdb_devices(device_ip=None, limit=100):
+    cmdb_url = HOST + "/phoenix/rest/cmdbDeviceInfo/devices"
+
+    if device_ip:
+        cmdb_url += "?includeIps=" + device_ip
+
+    response = requests.get(cmdb_url, verify=VERIFY_SSL, auth=AUTH)
+    list_of_devices = json.loads(xml2json(response.text))
+
+    if 'response' in list_of_devices:
+        return_error(list_of_devices["response"]["error"]["description"])
+    elif 'devices' in list_of_devices:
+        list_of_devices = list_of_devices['devices']['device']
+    elif 'device' in list_of_devices:
+        list_of_devices = [list_of_devices['device']]
+
+    return list_of_devices[:limit]
+
+
+def get_events_by_query(query, report_window="60", report_type="Minute", limit="20", extended_data='false',
+                        max_wait_time=60):
+    session = login()
```

Comment:
why login again?

Reply:
it's not again, this is a first (this command is called from the main)